src/coord/src/coord.rs

```diff
@@ -905,19 +945,17 @@ impl Coordinator {
                     // reduce the volume of the collection, but we don't have that
                     // information here.
                     if !index_state.upper.frontier().is_empty() {
-                        let mut compaction_frontier = Antichain::new();
-                        for time in index_state.upper.frontier().iter() {
-                            compaction_frontier.insert(
+                        // The since_handle for this GlobalId should have already been registered with
```

Comment:
This is a bit convoluted so please bear with me. If two materialized views depend or depended on some source, the compaction fronteer for the second view does not advance. Same happens if you drop and recreate a single view.

1. We use ```AS OF 0``` to see the current compaction frontier for a source. For example:
```
materialize=> select * from v1 as of 0;
ERROR:  Timestamp (0) is not valid for all inputs: [(User(366), Some(Antichain { elements: [1618563589623] }))]
```
2. If we issue such a query repeatedly against a base table, we see the frontier advancing in time, but not for second materialized views.

3. Lets use the following python script:
```
print("DROP SCHEMA public CASCADE;")
print("CREATE SCHEMA public;")
print("CREATE TABLE t1 (f1 INTEGER);")
print("CREATE MATERIALIZED VIEW v1 AS SELECT * FROM t1;")
print("CREATE MATERIALIZED VIEW v2 AS SELECT * FROM t1;")
for i in range(0,10000):
    print("SELECT COUNT(*) FROM v1 AS OF 0;")
for i in range(0,10000):
    print("SELECT COUNT(*) FROM v2 AS OF 0;")
```

4. and run it like this:

```
python ts.py | psql -p 6875 -h 127.0.0.1 -U materialize 2>&1 | sort | uniq -
```
5. We get the following output:
```
     13 ERROR:  Timestamp (0) is not valid for all inputs: [(User(404), Some(Antichain { elements: [1618564079768] }))]
   1552 ERROR:  Timestamp (0) is not valid for all inputs: [(User(404), Some(Antichain { elements: [1618564079797] }))]
   5032 ERROR:  Timestamp (0) is not valid for all inputs: [(User(404), Some(Antichain { elements: [1618564079999] }))]
     22 ERROR:  Timestamp (0) is not valid for all inputs: [(User(404), Some(Antichain { elements: [1618564080588] }))]
   3339 ERROR:  Timestamp (0) is not valid for all inputs: [(User(404), Some(Antichain { elements: [1618564080592] }))]
     42 ERROR:  Timestamp (0) is not valid for all inputs: [(User(404), Some(Antichain { elements: [1618564080999] }))]
```
This indicates that for view 404, we got 6 different frontiers as the test progressed, which is consistent with the frontier advancing over time.

6. However, we also get the following output:
```
  10000 ERROR:  Timestamp (0) is not valid for all inputs: [(User(406), Some(Antichain { elements: [1618564079743] }))]
```

Which means that the frontier for view 406 (the second materialized view) was frozen in time at position 1618564079743. 

7. On "main", multiple frontier values are reported for both views.

Reply:
Here as a testdrive test case:

```
> CREATE TABLE t1 (f1 INTEGER, f2 INTEGER);

> CREATE MATERIALIZED VIEW v1 AS SELECT * FROM t1;

> CREATE MATERIALIZED VIEW v2 AS SELECT * FROM t1;

# Sleep enough for the frontier to advance 

> SELECT mz_internal.mz_sleep(2);
<null>

# Check that we are not allowed to go back to a time before
# the frontier

! SELECT * FROM t1 AS OF NOW() - INTERVAL '1 second';
is not valid for all inputs

! SELECT * FROM v1 AS OF NOW() - INTERVAL '1 second';
is not valid for all inputs

! SELECT * FROM v2 AS OF NOW() - INTERVAL '1 second';
is not valid for all inputs
```

Passes on main, fails in this branch because the last SELECT succeeds

Reply:
This testdrive case seems to pass now!